### PR TITLE
[Component] Text areas

### LINKED ISSUE #300: [Component] TextArea
I'm envisioning this will live in a group alongside `Text input`
- Text inputs
  - Overview
  - \+ Text input
    - Enabled
    - Disabled
    - etc...
  - \+ Text area
    - Enabled
    - Disabled
    - etc...

## TODO
- [x] Implement [TextArea](https://cfpb.github.io/design-system/components/text-inputs)
- [x] Unit tests
- [x] Stories
  - Enabled
  - Disabled
  - Full width
  - Hover
  - Focus
  - Success
  - Warning
  - Error
- [x] Create shared Overview for `Text inputs`

### PR DESCRIPTION
**Pull request by meissadia**
Closes #300 

## Changes
- Implement [TextArea](https://cfpb.github.io/design-system/components/text-inputs) w/ unit tests
- Add `TextArea` to Storybook
  - Enabled
  - Disabled
  - Full width
  - Hover
  - Focus
  - Success
  - Warning
  - Error
- Create shared Overview for `Text inputs` (TextInput, TextArea)
- Address some Storybook sidebar organization issues

## How to test this PR

1. yarn test Alert
2. yarn test TextInput
3. [Preview deployment: Text inputs](https://deploy-preview-306--cfpb-design-system-react.netlify.app/?path=/docs/components-verified-text-inputs-overview--overview)

### TextArea stories
screencapture-localhost-6006-2024-02-07-11_18_26

### Organization of the sidebar
I noticed that the subfolders where we had manually created an Overview page (ex. Checkboxes) were being displayed out of order (shown first) instead of in alphabetical order.  Storybook does have some sort settings to customize this display, but the simple "sort alphabetically" did not correctly handle our case.  Ultimately I had to manually define the sort order but I think that's a small maintenance price to display things correctly.

|Before|Alphabetical sort|Manually defined sort|
|---|---|---|
|before|alphabetical|Screenshot 2024-02-07 at 10 07 43 AM|

### CODE REVIEW

**Comment by meissadia** [comment]
@natalia-fitzgerald The added TextArea stories are in the (Verified) section because they are added to `Text inputs` and TextInput was already verified.  Please let me know if you have time to verify TextArea against the DS.

**Review by natalia-fitzgerald — APPROVED** [review]
@meissadia This looks great! It's awesome to see all the states visualized. And thanks for the manual work you did to display these in the expected/correct order.

**Comment by meissadia** [comment]
### One last update to page headers
- Overview: Already has description + source link
  Screenshot 2024-02-07 at 5 10 38 PM
- TextInput: Remove description, keep source link
  Screenshot 2024-02-07 at 5 10 28 PM
- TextArea: Add source link
  Screenshot 2024-02-07 at 5 10 16 PM

File: src/components/TextInput/TextArea.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  onChange?: (value: string) => void;
+  className?: string;
+  isDisabled?: boolean;
+  isFullWidth?: boolean;
+  placeholder?: string;
+  status?: TextInputStatusType;
+}
+
+/**
+ * Source: <a href='https://cfpb.github.io/design-system/components/text-inputs' target='_blank'> https://cfpb.github.io/design-system/components/text-inputs</a>
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/components/TextInput/TextArea.tsx]: @shindigira Solid rec.  Updated.

**Pull request merged**